flake.nix

```diff
@@ -1,14 +1,41 @@
 {
-  description = "binplz";
+  description = "binplz-server";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs/release-22.05";
+  inputs.nixpkgs = { };
```

Comment:
What does it mean to have this be empty?  I tried looking in `nix flake --help`, but I couldn't really tell what Nix would do in this case.

Reply:
Oh sorry, you can ignore that. I haven't really looked at the parts that conflict with the current `master` yet. But yeah, when I rebase I'll keep it `release-22.05`.

Reply:
Oh, I don't really have a strong opinion what we pin to, but I'm just wondering how Nix interprets a flake reference that is empty.  So when you have `inputs.nixpkgs = {}`, how does Nix know where to find the `nixpkgs` input?  I don't think I've seen this before.

Reply:
> I'm just wondering how Nix interprets a flake reference that is empty

I think it's just the same as when you have an input that is only mentioned in the attr set argument to `outputs`. Like how this is a valid flake, when you `nix flake lock` it pulls the `nixpkgs` pin from your flake registry:
```nix
{ outputs = { self, nixpkgs}: { }; }
```

> Oh, I don't really have a strong opinion what we pin to

I have recently started of pinning all my `nixpkgs` flake inputs to a recent release, in hopes of getting more overlap with other flakes even if the pin is not exactly the same. Not sure yet if it actually works; my store still has 4 different GHC 9.0.2s.

Reply:
Oh nice, that makes sense.  Thanks!

> Not sure yet if it actually works; my store still has 4 different GHC 9.0.2s.

Only _4_??  Sounds like a huge success to me!!!